test/integration/spec/rtcpeerconnection.js

```diff
@@ -786,42 +735,133 @@ function makeStream(constraints) {
   return new Promise((resolve, reject) => getUserMedia(resolve, reject));
 }
 
-function testAddStream() {
+function testAddTrack() {
   var test;
   var stream;
 
-  before(() => {
-    return makeStream().then(_stream => stream = _stream);
+  before(async () => {
+    stream = await makeStream();
   });
 
-  beforeEach(() => {
-    return makeTest().then(_test => test = _test);
+  beforeEach(async () => {
+    test = await makeTest();
+  });
+
+  context('when the RTCPeerConnection is closed', () => {
+    var exception;
+
+    beforeEach(() => {
```

Comment:
Since we are using `beforeEach`, we should probably `null` out `exception` each time, especially if we anticipate adding more tests here.